networking/p2p/src/main/java/tech/pegasys/teku/networking/p2p/DiscoveryNetwork.java

```diff
@@ -70,6 +72,23 @@
   public void initialize() {
     setPreGenesisForkInfo();
     getEnr().ifPresent(StatusLogger.STATUS_LOG::listeningForDiscv5PreGenesis);
+
+    // Set connection manager peer predicate so that we don't attempt to connect peers with
+    // different fork digests
+    connectionManager.addPeerPredicate(
+        peer ->
+            enrForkId
+                .map(EnrForkId::getForkDigest)
+                .flatMap(
+                    forkDigest ->
+                        peer.getEnrForkId()
+                            .map(
+                                peerEnrForkId ->
+                                    SimpleOffsetSerializer.deserialize(
+                                            peerEnrForkId, EnrForkId.class)
+                                        .getForkDigest()
+                                        .equals(forkDigest)))
+                .orElse(false));
```

Comment:
nit: maybe extract this as a private method rather than doing it inline to avoid all the wrapping.

Reply:
Done.